torch/nn/parallel/data_parallel.py

```diff
@@ -142,6 +137,12 @@ def __init__(self, module, device_ids=None, output_device=None, dim=0):
     def forward(self, *inputs, **kwargs):
         if not self.device_ids:
             return self.module(*inputs, **kwargs)
+
+        if not all(t.is_cuda and t.device.index == self.device_ids[0]
+                   for t in chain(self.module.parameters(), self.module.buffers())):
+            raise RuntimeError("module must have its parameters and buffers "
```

Comment:
can you improve on this: "but found one of it's parameters on device: {}". I'd like to know also what I did wrong.

Reply:
Fixed!